src/prebid.js

```diff
@@ -105,6 +105,19 @@ function setRenderSize(doc, width, height) {
   if (doc.defaultView && doc.defaultView.frameElement) {
     doc.defaultView.frameElement.width = width;
     doc.defaultView.frameElement.height = height;
+    // in case of nested iframes, resize those as well
+    try {
+      if (doc.defaultView.parent !== window.top) {
+        setRenderSize(doc.defaultView.parent.document, width, height);
+      }
+    } catch (e) {
+      // x-domain try one more time.
```

Comment:
why would it work after it just failed?

Reply:
Sorry I should have made the comment more clear. This code is executed if there is an x-domain in the chain (because of throw we know this), however the most immediate parent may not be x-domain so it could potentially be resized. If it throws again we know we hit the x-domain so we are done

Reply:
but wouldn't it have already been resized the first time the function was called?  that's the first thing it does.

Also, I ran a test in chrome and I don't ever see an error thrown.  When there's a x-domain iframe the `if (doc.defaultView && doc.defaultView.frameElement) {` condition fails and the function exits silently.  Maybe this would work differently in other/older browsers?

Reply:
oh interesting. I'll look more into this. Thanks

Reply:
so the case this catches is for window => x-domain iframe => friendly Iframe => ad document iframe => ad

So the first pass will resize the ad document iframe. Then it will check against if `parent !== window.top` this will throw right away because of the x-domain iframe in the chain, however, `doc.default.view.parent` is pointing to a friendly iframe so it sends that through as the next resize option (`doc.defaultView.parent.document`). Hopefully that's clear.